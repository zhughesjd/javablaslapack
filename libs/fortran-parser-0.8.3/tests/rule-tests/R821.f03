! Test select-type-construct
!      select-type-construct  is  select-type-stmt
!                                   [ type-guard-stmt 
!                                       block ] ...
!                                 end-select-type-stmt
!
!      select-type-stmt  is  [ select-construct-name : ] SELECT TYPE 
!                              ( [ associate-name => ] selector)
!
!      type-guard-stmt  is  TYPE IS ( type-spec) [ select-construct-name ]
!                       or  CLASS IS ( type-spec ) [ select-construct-name ]
!                       or  CLASS DEFAULT [ select-construct-name ]
!
!      end-select-type-stmt  is  END SELECT [ select-construct-name ]
!
! Tested here: select-type-construct, select-type-stmt, type-guard-stmt, and 
! end-select-type-stmt.
!
! Not tested here: block, select-construct-name, associate-name, selector, 
! and type-spec
select type (a => p) 
class is ( point )
   x = 1
type is (point_3d)
   x = 2
class default
   x = 3
end select 

select type (p) 
class is ( point )
   x = 1
type is (point_3d)
   x = 2
class default
   x = 3
end select 


my_select: select type (p) 
class is ( point ) my_select
   x = 1
type is (point_3d) my_select
   x = 2
class default my_select
   x = 3
end select my_select

end


