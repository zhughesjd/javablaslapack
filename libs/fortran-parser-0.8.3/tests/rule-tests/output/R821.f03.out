*******************************************
args[1]: R821.f03
R1101:main-program__begin:
R 204:specification-part: numUseStmts=0 numImportStmts=0 numDeclConstructs=0
R 822:select-type:
R 613:part-ref: id=[@49,889:889='p',<209>,21:18] hasSelectionSubscriptList=false hasImageSelector=false
R 612:data-ref: numPartRef=1
R -3:designator-or-func-ref:
R 701:primary:
R 819:selector:
R 822:select-type-stmt: selectConstructName=null associateName=[@45,884:884='a',<209>,21:13]
R 455:derived-type-spec: typeName=[@59,904:908='point',<209>,22:11] hasTypeParamSpecList=false
R 401:type-spec:
R 823:type-guard-stmt: selectConstructName=null
R 613:part-ref: id=[@66,915:915='x',<209>,23:3] hasSelectionSubscriptList=false hasImageSelector=false
R 612:data-ref: numPartRef=1
R 603:designator: hasSubstringRange=false
R 601:variable:
R 406:int-literal-constant: digitString=[@70,919:919='1',<10>,23:7] kindParam=null
R 306:literal-constant:
R 701:primary:
R 734:assignment-stmt:
R 214:action-stmt:
R 213:executable-construct:
R 209:execution-part-construct:
R 801:block:
R 455:derived-type-spec: typeName=[@77,930:937='point_3d',<209>,24:9] hasTypeParamSpecList=false
R 401:type-spec:
R 823:type-guard-stmt: selectConstructName=null
R 613:part-ref: id=[@83,943:943='x',<209>,25:3] hasSelectionSubscriptList=false hasImageSelector=false
R 612:data-ref: numPartRef=1
R 603:designator: hasSubstringRange=false
R 601:variable:
R 406:int-literal-constant: digitString=[@87,947:947='2',<10>,25:7] kindParam=null
R 306:literal-constant:
R 701:primary:
R 734:assignment-stmt:
R 214:action-stmt:
R 213:executable-construct:
R 209:execution-part-construct:
R 801:block:
R 823:type-guard-stmt: selectConstructName=null
R 613:part-ref: id=[@96,966:966='x',<209>,27:3] hasSelectionSubscriptList=false hasImageSelector=false
R 612:data-ref: numPartRef=1
R 603:designator: hasSubstringRange=false
R 601:variable:
R 406:int-literal-constant: digitString=[@100,970:970='3',<10>,27:7] kindParam=null
R 306:literal-constant:
R 701:primary:
R 734:assignment-stmt:
R 214:action-stmt:
R 213:executable-construct:
R 209:execution-part-construct:
R 801:block:
R 824:end-select-type-stmt: id=null
R 821:select-type-construct:
R 213:executable-construct:
R 822:select-type:
R 613:part-ref: id=[@113,998:998='p',<209>,30:13] hasSelectionSubscriptList=false hasImageSelector=false
R 612:data-ref: numPartRef=1
R -3:designator-or-func-ref:
R 701:primary:
R 819:selector:
R 822:select-type-stmt: selectConstructName=null associateName=null
R 455:derived-type-spec: typeName=[@123,1013:1017='point',<209>,31:11] hasTypeParamSpecList=false
R 401:type-spec:
R 823:type-guard-stmt: selectConstructName=null
R 613:part-ref: id=[@130,1024:1024='x',<209>,32:3] hasSelectionSubscriptList=false hasImageSelector=false
R 612:data-ref: numPartRef=1
R 603:designator: hasSubstringRange=false
R 601:variable:
R 406:int-literal-constant: digitString=[@134,1028:1028='1',<10>,32:7] kindParam=null
R 306:literal-constant:
R 701:primary:
R 734:assignment-stmt:
R 214:action-stmt:
R 213:executable-construct:
R 209:execution-part-construct:
R 801:block:
R 455:derived-type-spec: typeName=[@141,1039:1046='point_3d',<209>,33:9] hasTypeParamSpecList=false
R 401:type-spec:
R 823:type-guard-stmt: selectConstructName=null
R 613:part-ref: id=[@147,1052:1052='x',<209>,34:3] hasSelectionSubscriptList=false hasImageSelector=false
R 612:data-ref: numPartRef=1
R 603:designator: hasSubstringRange=false
R 601:variable:
R 406:int-literal-constant: digitString=[@151,1056:1056='2',<10>,34:7] kindParam=null
R 306:literal-constant:
R 701:primary:
R 734:assignment-stmt:
R 214:action-stmt:
R 213:executable-construct:
R 209:execution-part-construct:
R 801:block:
R 823:type-guard-stmt: selectConstructName=null
R 613:part-ref: id=[@160,1075:1075='x',<209>,36:3] hasSelectionSubscriptList=false hasImageSelector=false
R 612:data-ref: numPartRef=1
R 603:designator: hasSubstringRange=false
R 601:variable:
R 406:int-literal-constant: digitString=[@164,1079:1079='3',<10>,36:7] kindParam=null
R 306:literal-constant:
R 701:primary:
R 734:assignment-stmt:
R 214:action-stmt:
R 213:executable-construct:
R 209:execution-part-construct:
R 801:block:
R 824:end-select-type-stmt: id=null
R 821:select-type-construct:
R 213:executable-construct:
R 209:execution-part-construct:
R 822:select-type:
R 613:part-ref: id=[@181,1119:1119='p',<209>,40:24] hasSelectionSubscriptList=false hasImageSelector=false
R 612:data-ref: numPartRef=1
R -3:designator-or-func-ref:
R 701:primary:
R 819:selector:
R 822:select-type-stmt: selectConstructName=[@173,1095:1103='my_select',<209>,40:0] associateName=null
R 455:derived-type-spec: typeName=[@191,1134:1138='point',<209>,41:11] hasTypeParamSpecList=false
R 401:type-spec:
R 823:type-guard-stmt: selectConstructName=[@195,1142:1150='my_select',<209>,41:19]
R 613:part-ref: id=[@200,1155:1155='x',<209>,42:3] hasSelectionSubscriptList=false hasImageSelector=false
R 612:data-ref: numPartRef=1
R 603:designator: hasSubstringRange=false
R 601:variable:
R 406:int-literal-constant: digitString=[@204,1159:1159='1',<10>,42:7] kindParam=null
R 306:literal-constant:
R 701:primary:
R 734:assignment-stmt:
R 214:action-stmt:
R 213:executable-construct:
R 209:execution-part-construct:
R 801:block:
R 455:derived-type-spec: typeName=[@211,1170:1177='point_3d',<209>,43:9] hasTypeParamSpecList=false
R 401:type-spec:
R 823:type-guard-stmt: selectConstructName=[@214,1180:1188='my_select',<209>,43:19]
R 613:part-ref: id=[@219,1193:1193='x',<209>,44:3] hasSelectionSubscriptList=false hasImageSelector=false
R 612:data-ref: numPartRef=1
R 603:designator: hasSubstringRange=false
R 601:variable:
R 406:int-literal-constant: digitString=[@223,1197:1197='2',<10>,44:7] kindParam=null
R 306:literal-constant:
R 701:primary:
R 734:assignment-stmt:
R 214:action-stmt:
R 213:executable-construct:
R 209:execution-part-construct:
R 801:block:
R 823:type-guard-stmt: selectConstructName=[@229,1213:1221='my_select',<209>,45:14]
R 613:part-ref: id=[@234,1226:1226='x',<209>,46:3] hasSelectionSubscriptList=false hasImageSelector=false
R 612:data-ref: numPartRef=1
R 603:designator: hasSubstringRange=false
R 601:variable:
R 406:int-literal-constant: digitString=[@238,1230:1230='3',<10>,46:7] kindParam=null
R 306:literal-constant:
R 701:primary:
R 734:assignment-stmt:
R 214:action-stmt:
R 213:executable-construct:
R 209:execution-part-construct:
R 801:block:
R 824:end-select-type-stmt: id=[@244,1243:1251='my_select',<209>,47:11]
R 821:select-type-construct:
R 213:executable-construct:
R 209:execution-part-construct:
R 208:execution-part:
R1103:end-program-stmt: id=null
R1101:main-program: hasProgramStmt=false hasExecutionPart=true hasInternalSubprogramPart=false
Parser exiting normally
********************************************
